### DISCUSSION TITLE
API - Refreshing Access Token - 400 Bad Request - cloudflare

### DISCUSSION THREAD

**Discussion by amypgon** [supabase/supabase#13200] [Questions]
**Request via CURL in cli**
`curl --location --request POST 'https://(REDACTED).supabase.co/auth/v1/token?grant_type=refresh_token' \
--header 'apikey: (REDACTED)' \
--header 'Host: (REDACTED).supabase.co' \
--header 'content-length: 43' \
--header 'Content-Type: application/json' \
--data-raw '{"refresh_token": "(REDACTED)"}'`
Response
`{
    "error": "invalid_grant",
    "error_description": "Invalid Refresh Token"
}`

**Request via curl in code**
`(SAME REQUEST AS ABOVE`
Response
`html>
head>title>400 Bad Request/title>/head>
body>
center>h1>400 Bad Request/h1>/center>
hr><center>cloudflare</center>
/body>
/html>`

**Could you please provide an explanation of potential resolution of this issue?**

**Comment by RyanCarterTheDev** [comment]
Did you ever figure this out? I'm experiencing the same.

**Comment by Detzler** [comment]
@amypgon @RyanCarterTheDev 
Got a similar problem when posting json data through Cloudflare. How did you fix yours?
